geopandas/tests/test_plotting.py

```diff
@@ -1433,6 +1436,73 @@ def test_polygons_values(self):
         ax.cla()
 
 
+@pytest.mark.skipif(not compat.PANDAS_GE_025, reason="requires pandas > 0.24")
+class TestGeoplotAccessor:
+    def setup_method(self):
+        geometries = [Polygon([(0, 0), (1, 0), (1, 1)]), Point(1, 3)]
+        x = [1, 2]
+        y = [10, 20]
+        self.gdf = GeoDataFrame({"geometry": geometries, "x": x, "y": y})
+        self.df = pd.DataFrame({"x": x, "y": y})
+
+    def compare_figures(self, kind, fig_test, fig_ref, kwargs):
+        """Compare Figures."""
+        ax_pandas_1 = self.df.plot(kind=kind, **kwargs)
+        ax_geopandas_1 = self.gdf.plot(kind=kind, **kwargs)
+        fig_test.subplots().plot(
+            ax=ax_pandas_1,
+        )
+        fig_ref.subplots().plot(ax=ax_geopandas_1)
+
+        ax_pandas_2 = getattr(self.df.plot, kind)(**kwargs)
+        ax_geopandas_2 = getattr(self.gdf.plot, kind)(**kwargs)
+        fig_test.subplots().plot(ax=ax_pandas_2)
+        fig_ref.subplots().plot(ax=ax_geopandas_2)
+
+    @check_figures_equal()
+    def test_pandas_kind(self, fig_test, fig_ref):
+        """Test Pandas kind."""
+        import importlib
+        from geopandas.plotting import GeoplotAccessor
+
+        _pandas_kinds = GeoplotAccessor._pandas_kinds
+        _scipy_dependent_kinds = ["kde", "density"]  # Needs scipy
+        _y_kinds = ["pie"]  # Needs y
+        _xy_kinds = ["scatter", "hexbin"]  # Needs x & y
+        kwargs = {}
+        for kind in _pandas_kinds:
```

Comment:
Would it be possible to turn this loop over kinds into `@pytest.mark.parametrize`? That way, if one of the options fails we can directly from pytest see which. (It is not critical though).

You do `@pytest.mark.parametrize("kind", GeoplotAccessor._pandas_kinds)` and just remove this row (I think).

Reply:
Ah makes sense ! pytest is really neat